--- a/src/Serializer/Subscriber/LanguageIsoSubscriber.php
+++ b/src/Serializer/Subscriber/LanguageIsoSubscriber.php
@@ -70,7 +70,7 @@ class LanguageIsoSubscriber implements EventSubscriberInterface
     public function onPreDeserialize(PreDeserializeEvent $event): void
     {
         $data = $event->getData();
-        if (\is_array($data) && isset($data['languageISO']) && !isset($data['languageIso'])) {
+        if (\is_array($data) && isset($data['languageISO']) && \trim((string)$data['languageISO']) !== '' && !isset($data['languageIso'])) {
             $language            = $this->languages->fromISO_639_2b($data['languageISO']);
             $data['languageIso'] = $language->toISO_639_1();
             $event->setData($data);
